pkg/continuoustest/manager.go

```diff
@@ -92,5 +93,9 @@ func (m *Manager) Run(ctx context.Context) error {
 		})
 	}
 
-	return group.Wait()
+	err := group.Wait()
+	if m.cfg.SmokeTest {
```

Comment:
This means we'll never return an actual error in smoke test mode so `mimir-continuous-test` or `mimir -module=continuous-test` will never exit with a non-zero code. Is that actually what we want?

Reply:
Yeah I was contemplating that. If smokes fail, we still want to exit. The errorgroup returns the first non-nil error of all of them (right now there's only 1 test). It will output the failures to all the tests, but only return the first. So in that scenario I was thinking "what's the value of returning multiple errors or a conjoined error?" and I think the answer is only that you would get the first test failure along with the module stop error. So I'm comfortable with it going the route of "output all of them" (again right now there's only 1), but return a successful _run_ of the smoke tests? It's not the most comfortable, but given the return error to stop module hack, that was my thought process. opinions?

Reply:
The alternative is returning all the errors and checking them all. Which we could do. But still breaks down at the way the errorgroup works. I didn't know that there was enough value here to do that.

Reply:
That sounds fine to me